jbi/services/jira.py

```diff
@@ -238,7 +257,8 @@ def get_issue(self, context: ActionContext, issue_key):
         """Return the Jira issue fields or `None` if not found."""
         logger.debug("Getting issue %s", issue_key, extra=context.model_dump())
         try:
-            response = self.client.get_issue(issue_key)
+            with self.client.rid(context.rid):
+                response = self.client.get_issue(issue_key)
```

Comment:
I was wondering whether we could have something less verbose instead of duplicating the usage of the context manager...

```py
def client(self, rid):
    with self._client.rid(rid):
        return self._client
```

```py
response = self.client(rid).get_issue(issue_key)
```

Reply:
I like that API too (`self.client(rid).<method>`), but I think returning the client from within the context manager will immediately trigger the `__exit__` method, right? That is:
```
client = self.client(123)
```

Will set `self._client._rid = 123`, return the client, then immediately set `self._client._rid = None` before we use it. I think we'll still need a bunch of `with` blocks wrapping actual method calls one way or another.

Reply:
oh you're right 😵

Reply:
only way would be to clone on each call to `self.client(rid)`